aten/src/ATen/BatchingRegistrations.cpp

```diff
@@ -56,7 +56,36 @@ Tensor sum_batching_rule(const Tensor& self, IntArrayRef dims, bool keepdim, opt
 }
 
 Tensor mul_batching_rule(const Tensor& self, const Tensor& other) {
-  auto physical_args = BroadcastingVmapTransform::logicalToPhysical({self, other});
+  if (self.dim() > 0 && other.dim() > 0) {
+    // Fast path: any type promotion is handled inside at::mul
+    auto physical_args = BroadcastingVmapTransform::logicalToPhysical({self, other});
+    auto result = at::mul(physical_args[0].tensor(), physical_args[1].tensor());
+    return physical_args[0].newLogicalFromPhysical(result);
+  }
+
+  // Handle type promotion of logically scalar tensors.
```

Comment:
HMMM. At least for these binary operations, can't you just NOT view the scalar, and pass it in directly? At least in those cases, the normal broadcasting behavior of the operation will work properly, AND you'll have cross-device scalars working correctly too (well, correctly as in, however TensorIterator handles them today).

Reply:
There are two cases here:

Case 1: multiplication between Tensor and Scalar
`vmap(torch.mul, (0, None))(torch.rand(B0), 1)`
In this case, the strategy of not viewing the scalar and passing it in directly works.

Case 2: multiplication between Tensor and a "logically Scalar Tensor"
`vmap(torch.mul)(torch.rand(B0, 3, dtype=torch.float), torch.rand(B0, dtype=torch.double))`
In this case, the strategy of not viewing the scalar and passing it in directly doesn't work. That's because there isn't a "physical scalar" to be viewed here. 

I could split up those two cases so that the cross-device-scalars work for case 1. However, we'll still need some manual type-promotion code to handle case 2 (or we can just ignore case 2 completely for now and mark it as a known issue). What do you think?

Reply:
I haven't worked out all the details, but it seems to me like the distinguishing characteristic between (1) and (2) is whether or not `is_wrapped_number()` is true or not on the Tensor.

Reply:
I updated the code to distinguish between case (1) (plus another case that I'll call "case 1.5") and case (2).

In case (2) cross-device scalars still don't work, but it should get encountered less often than case (1)